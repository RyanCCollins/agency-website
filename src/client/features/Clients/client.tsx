import * as React from 'react';
import { Box, Image } from 'ui';

export default function Client({
  image,
  name,
}) {
  return (
    <Box style={{ flexBasis: '33%' }} pad="large" alignItems="center">
      <Image alt={`${name} logo`} src={image} imageSize="small" />
    </Box>
  );
}
