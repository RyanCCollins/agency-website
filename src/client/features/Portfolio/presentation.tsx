import * as React from 'react';
import { Box, Headline } from 'ui';
import { CaseStudies } from 'components';
import { Props } from './types';
import Hr from './hr';

export default function Presentation({
  theme,
  isMounted,
  data: { items },
}: Props) {
  return (
    <Box alignItems="center" backgroundColor={theme.offwhite}>
      <Box pad="large" alignItems="center">
        <Headline color="#666" fontWeight={700}>Case Studies</Headline>
        <Hr color="#666" />
      </Box>
      <CaseStudies isMounted={isMounted} items={items} />
    </Box>
  );
}
