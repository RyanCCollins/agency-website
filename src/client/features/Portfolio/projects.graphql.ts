import gql from 'graphql-tag';

export default gql`
  query Projects {
    projects {
      id: _id
      title
      image
      content
    }
  }
`;
