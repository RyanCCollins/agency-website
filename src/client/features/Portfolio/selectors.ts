import { State as GlobalState } from '../../state';
import { State, ErrorType } from './types';
import { createSelector, Selector } from 'reselect';
import { Data } from './types';

export const selectPortfolio = () => (state: GlobalState): State => state.portfolio;

export type SelectIsLoading = Selector<GlobalState, boolean>;
export const selectIsLoading: SelectIsLoading = createSelector(
  selectPortfolio(),
  (portfolio) => portfolio.isLoading,
);

export type SelectError = Selector<GlobalState, ErrorType>;
export const selectError: SelectError = createSelector(
  selectPortfolio(),
  (portfolio) => portfolio.error,
);

export type SelectData = Selector<GlobalState, Data>;
export const selectData: SelectData = createSelector(
  selectPortfolio(),
  (portfolio) => portfolio.data,
);
