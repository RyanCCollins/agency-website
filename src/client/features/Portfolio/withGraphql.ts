import { graphql } from 'react-apollo';
import PROJECTS_QUERY from './projects.graphql';

export default graphql(PROJECTS_QUERY, {
  props: ({ data: { loading, projects, error } }) => ({
    loading,
    projects,
    error,
  }),
});
