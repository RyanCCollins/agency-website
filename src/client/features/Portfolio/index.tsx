import * as React from 'react';
import { withTheme } from 'styled-components';
import Presentation from './presentation';
import { ThemeColorMap } from '../../types';
import withGraphql from './withGraphql';

interface State {
  isMounted: boolean,
}

export interface Project {
  title: string;
  image: string;
}

export interface Props {
  theme: ThemeColorMap;
  projects: Project[];
  loading: boolean;
  error: string;
}

class Portfolio extends React.Component<Props, State> {
  constructor() {
    super();
    this.state = {
      isMounted: false,
    };
  }
  public componentDidMount() {
    setTimeout(() => {
      this.setState({
        isMounted: true,
      });
    }, 1000);
  }
  public render() {
    return (
      <Presentation {...this.state} {...this.props} />
    );
  }
}

export default withTheme(withGraphql(Portfolio));
