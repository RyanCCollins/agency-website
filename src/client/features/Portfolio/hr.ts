import styled from 'styled-components';
import styles from './styles'; 

export interface Props {
  color: string;
}

export default styled.div`
  ${(props: Props) => styles(props)}
`;
