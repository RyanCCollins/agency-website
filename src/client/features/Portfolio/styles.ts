import { css } from 'styled-components';
import { Props } from './hr'; 

export default function styles({ color }: Props) {
  return css`
    background-color: ${color};
    width: 100px;
    height: 3px;
    display: block;
    margin: 35px 0;
    margin-right: auto;
    margin-left: auto;
  `;
}
