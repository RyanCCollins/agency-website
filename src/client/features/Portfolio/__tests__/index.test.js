import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import { ThemeProvider } from 'styled-components';
import { ApolloProvider } from 'react-apollo';
import React from 'react';
import Portfolio from '../index';
import theme from '../../../theming';
import client from '../../../apolloClient';
import store from '../../../store';

describe('<Portfolio />', () => {

  describe('when shallow rendered', () => {
    const wrapper = mount(
      <ApolloProvider store={store} client={client}>
        <ThemeProvider theme={theme}>
          <Portfolio />
        </ThemeProvider>
      </ApolloProvider>
    );

    it('should create a snapshot', () => {
      expect(mountToJson(wrapper)).toMatchSnapshot();
    });
  });
});
