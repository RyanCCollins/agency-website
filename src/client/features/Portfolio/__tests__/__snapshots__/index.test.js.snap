// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Portfolio /> when shallow rendered should create a snapshot 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "addTypename": true,
      "dataId": undefined,
      "fieldWithArgs": [Function],
      "initialState": Object {},
      "middleware": [Function],
      "mutate": [Function],
      "networkInterface": HTTPFetchNetworkInterface {
        "_afterwares": Array [],
        "_middlewares": Array [],
        "_opts": Object {},
        "_uri": "http://0.0.0.0:1338/api",
      },
      "query": [Function],
      "queryDeduplication": false,
      "queryManager": QueryManager {
        "addTypename": true,
        "deduplicator": Deduplicator {
          "inFlightRequestPromises": Object {},
          "networkInterface": HTTPFetchNetworkInterface {
            "_afterwares": Array [],
            "_middlewares": Array [],
            "_opts": Object {},
            "_uri": "http://0.0.0.0:1338/api",
          },
        },
        "fetchQueryPromises": Object {
          "2": Object {
            "promise": undefined,
            "reject": [Function],
            "resolve": [Function],
          },
        },
        "idCounter": 3,
        "networkInterface": HTTPFetchNetworkInterface {
          "_afterwares": Array [],
          "_middlewares": Array [],
          "_opts": Object {},
          "_uri": "http://0.0.0.0:1338/api",
        },
        "observableQueries": Object {
          "1": Object {
            "observableQuery": ObservableQuery {
              "isCurrentlyPolling": false,
              "observers": Array [
                Object {
                  "error": [Function],
                  "next": [Function],
                },
              ],
              "options": Object {
                "metadata": Object {
                  "reactComponent": Object {
                    "displayName": "Apollo(Portfolio)",
                  },
                },
                "query": Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "Projects",
                      },
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": null,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "projects",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "_id",
                                  },
                                  "selectionSet": null,
                                },
                                Object {
                                  "alias": null,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "title",
                                  },
                                  "selectionSet": null,
                                },
                                Object {
                                  "alias": null,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "image",
                                  },
                                  "selectionSet": null,
                                },
                                Object {
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "__typename",
                                  },
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 83,
                    "source": Object {
                      "body": "
      query Projects {
        projects {
          id: _id
          title
          image
        }
      }
    ",
                      "name": "GraphQL",
                    },
                    "start": 0,
                  },
                },
              },
              "queryId": "1",
              "queryManager": [Circular],
              "scheduler": QueryScheduler {
                "inFlightQueries": Object {},
                "intervalQueries": Object {},
                "pollingTimers": Object {},
                "queryManager": [Circular],
                "registeredQueries": Object {},
              },
              "shouldSubscribe": true,
              "subscriberFunction": [Function],
              "subscriptionHandles": Array [],
              "variables": Object {},
            },
          },
        },
        "pollingTimers": Object {},
        "queryDeduplication": false,
        "queryDocuments": Object {
          "1": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "Projects",
                },
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": null,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "projects",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "_id",
                            },
                            "selectionSet": null,
                          },
                          Object {
                            "alias": null,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "title",
                            },
                            "selectionSet": null,
                          },
                          Object {
                            "alias": null,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "image",
                            },
                            "selectionSet": null,
                          },
                          Object {
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "__typename",
                            },
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 83,
              "source": Object {
                "body": "
      query Projects {
        projects {
          id: _id
          title
          image
        }
      }
    ",
                "name": "GraphQL",
              },
              "start": 0,
            },
          },
        },
        "queryIdsByName": Object {
          "Projects": Array [
            "1",
          ],
        },
        "queryListeners": Object {
          "1": Array [
            [Function],
          ],
        },
        "reducerConfig": Object {
          "customResolvers": undefined,
          "dataIdFromObject": undefined,
        },
        "reduxRootSelector": [Function],
        "resultComparator": undefined,
        "resultTransformer": undefined,
        "scheduler": QueryScheduler {
          "inFlightQueries": Object {},
          "intervalQueries": Object {},
          "pollingTimers": Object {},
          "queryManager": [Circular],
          "registeredQueries": Object {},
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
        },
      },
      "reducerConfig": Object {
        "customResolvers": undefined,
        "dataIdFromObject": undefined,
      },
      "reduxRootKey": "apollo",
      "reduxRootSelector": null,
      "resetStore": [Function],
      "resultComparator": undefined,
      "resultTransformer": undefined,
      "setStore": [Function],
      "shouldForceFetch": false,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
      },
      "version": "local",
      "watchQuery": [Function],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "black1": "#0a0a0a",
        "black2": "#2d2d2d",
        "black3": "#555555",
        "dark1": "#13293D",
        "dark2": "#006494",
        "dark3": "#92374D",
        "error": "#ff324d",
        "gray1": "#696969",
        "gray2": "#7c7c7c",
        "gray3": "#828282",
        "light1": "#0072BB",
        "light2": "#ff9800",
        "light3": "#0C7489",
        "offwhite": "#f5f5f5",
        "ok": "#8cc800",
        "primary": "#03A9F4",
        "secondary": "#9c27b0",
        "warning": "#ffd602",
        "white1": "#fff",
        "white2": "#f3f3f3",
        "white3": "#e6e8ec",
      }
    }
  >
    <_class>
      <Apollo(Portfolio)
        theme={
          Object {
            "black1": "#0a0a0a",
            "black2": "#2d2d2d",
            "black3": "#555555",
            "dark1": "#13293D",
            "dark2": "#006494",
            "dark3": "#92374D",
            "error": "#ff324d",
            "gray1": "#696969",
            "gray2": "#7c7c7c",
            "gray3": "#828282",
            "light1": "#0072BB",
            "light2": "#ff9800",
            "light3": "#0C7489",
            "offwhite": "#f5f5f5",
            "ok": "#8cc800",
            "primary": "#03A9F4",
            "secondary": "#9c27b0",
            "warning": "#ffd602",
            "white1": "#fff",
            "white2": "#f3f3f3",
            "white3": "#e6e8ec",
          }
        }
      >
        <Portfolio
          loading={true}
          theme={
            Object {
              "black1": "#0a0a0a",
              "black2": "#2d2d2d",
              "black3": "#555555",
              "dark1": "#13293D",
              "dark2": "#006494",
              "dark3": "#92374D",
              "error": "#ff324d",
              "gray1": "#696969",
              "gray2": "#7c7c7c",
              "gray3": "#828282",
              "light1": "#0072BB",
              "light2": "#ff9800",
              "light3": "#0C7489",
              "offwhite": "#f5f5f5",
              "ok": "#8cc800",
              "primary": "#03A9F4",
              "secondary": "#9c27b0",
              "warning": "#ffd602",
              "white1": "#fff",
              "white2": "#f3f3f3",
              "white3": "#e6e8ec",
            }
          }
        >
          <Presentation
            isMounted={false}
            loading={true}
            theme={
              Object {
                "black1": "#0a0a0a",
                "black2": "#2d2d2d",
                "black3": "#555555",
                "dark1": "#13293D",
                "dark2": "#006494",
                "dark3": "#92374D",
                "error": "#ff324d",
                "gray1": "#696969",
                "gray2": "#7c7c7c",
                "gray3": "#828282",
                "light1": "#0072BB",
                "light2": "#ff9800",
                "light3": "#0C7489",
                "offwhite": "#f5f5f5",
                "ok": "#8cc800",
                "primary": "#03A9F4",
                "secondary": "#9c27b0",
                "warning": "#ffd602",
                "white1": "#fff",
                "white2": "#f3f3f3",
                "white3": "#e6e8ec",
              }
            }
          >
            <Box
              alignItems="center"
              backgroundColor="#f5f5f5"
            >
              <styled.div
                alignItems="center"
                backgroundColor="#f5f5f5"
              >
                <div
                  className="fCdVIs"
                >
                  <Box
                    alignItems="center"
                    pad="large"
                  >
                    <styled.div
                      alignItems="center"
                      pad="large"
                    >
                      <div
                        className="fDxjjW"
                      >
                        <Headline
                          color="#666"
                          fontSize="medium"
                          fontWeight={700}
                          margin="medium"
                          textAlign="center"
                        >
                          <styled.h1
                            color="#666"
                            fontSize="medium"
                            fontWeight={700}
                            margin="medium"
                            textAlign="center"
                          >
                            <h1
                              className="hLOqrJ"
                              color="#666"
                              fontSize="medium"
                              fontWeight={700}
                            >
                              Case Studies
                            </h1>
                          </styled.h1>
                        </Headline>
                        <styled.div
                          color="#666"
                        >
                          <div
                            className="gJIgCc"
                            color="#666"
                          />
                        </styled.div>
                      </div>
                    </styled.div>
                  </Box>
                </div>
              </styled.div>
            </Box>
          </Presentation>
        </Portfolio>
      </Apollo(Portfolio)>
    </_class>
  </ThemeProvider>
</ApolloProvider>
`;
